src/pydrex/visualisation.py

```diff
@@ -234,21 +241,22 @@ def pathline_box2d(
             **kwargs,
         )
 
-    P = np.asarray([[p[horizontal], p[vertical]] for p in positions])
-    if cpo_vectors is not None:
-        if cpo_strengths is None:
-            cpo_strengths = np.full(len(cpo_vectors), 1.0)
-        C = np.asarray(
-            [
-                f * np.asarray([c[horizontal], c[vertical]])
-                for f, c in zip(cpo_strengths, cpo_vectors, strict=True)
-            ]
-        )
-        cpo = ax.quiver(
-            P[:, 0],
-            P[:, 1],
-            C[:, 0],
-            C[:, 1],
+    dummy_dim = ({0, 1, 2} - set(_geo.to_indices2d(*ref_axes))).pop()
+    xi_2D = np.asarray([_utils.remove_dim(p, dummy_dim) for p in positions])
+    qcoll: plt.Quiver | plt.PathCollection
```

Comment:
Is that a type annotation in the middle of nowhere?

Reply:
It's valid syntax and covers the dual use of `qcoll` which subsequently collects either the output of `ax.quiver` or `ax.scatter`. Should be transparent to anything but mypy. Type hints now became more generic and less painfull (3.12): https://www.andy-pearce.com/blog/posts/2023/Dec/whats-new-in-python-312-type-hint-improvements/ and I'm starting to see the appeal, although I refuse to import `typing` anywhere so I'm not about to add lots of hints, just here and there to help with code review.